libraries/mspa/activityControls.js

```diff
@@ -0,0 +1,70 @@
+import {registerActivityControl} from '../../src/activities/rules.js';
+import {
+  ACTIVITY_ENRICH_EIDS,
+  ACTIVITY_ENRICH_UFPD,
+  ACTIVITY_SYNC_USER,
+  ACTIVITY_TRANSMIT_PRECISE_GEO
+} from '../../src/activities/activities.js';
+import {gppDataHandler} from '../../src/adapterManager.js';
+
+// default interpretation for MSPA consent(s):
+// https://docs.google.com/document/d/1yPEVx3bBdSkIw-QNkQR5qi1ztmn9zoXk-LaGQB6iw7Q
+
+function isBasicConsentDenied(cd) {
+  return ['MspaServiceProviderMode', 'Gpc'].some(prop => cd[prop] === 1) ||
+    cd.PersonalDataConsents === 2 ||
+    cd.KnownChildSensitiveDataConsents[0] === 1 || cd.KnownChildSensitiveDataConsents[1] !== 0;
+}
+
+function isSensitiveConsentDenied(cd) {
+  return ['SensitiveDataProcessingOptOutNotice', 'SensitiveDataLimitUseNotice'].some(prop => cd[prop] === 2)
+}
+
+function isConsentDenied(cd) {
+  return isBasicConsentDenied(cd) ||
+    ['SaleOptOut', 'SharingOptOut', 'TargetedAdvertisingOptOut'].some(prop => cd[prop] === 1) ||
+    ['SaleOptOutNotice', 'SharingNotice', 'SharingOptOutNotice', 'TargetedAdvertisingOptOutNotice'].some(prop => cd[prop] === 2);
+}
+
+function isTransmitUfpdConsentDenied(cd) {
+  return isConsentDenied(cd) ||
+    isSensitiveConsentDenied(cd) ||
+    cd.SensitiveDataProcessing.some((val, i) => (i < 6 || i > 7) && val === 1) ||
+    cd.SensitiveDataProcessing[6] !== 0;
+}
+
+function isTransmitGeoConsentDenied(cd) {
+  return isBasicConsentDenied(cd) ||
+    isSensitiveConsentDenied(cd) ||
+    cd.SensitiveDataProcessing[7] === 1
+}
+
+const CONSENT_RULES = {
+  [ACTIVITY_SYNC_USER]: isConsentDenied,
+  [ACTIVITY_ENRICH_EIDS]: isConsentDenied,
+  [ACTIVITY_ENRICH_UFPD]: isConsentDenied,
+  [ACTIVITY_ENRICH_UFPD]: isTransmitUfpdConsentDenied,
```

Comment:
Is this supposed to share the previous keyname?  

It looks like the `isConsentDenied` function for the `enrichUfpd` key is getting eaten for the 2nd value when this code is actually running (unless I'm missing something).

Reply:
Looks like we should just delete line 45, as the function called on line 46 incorporates that logic on line 30